tests/async_tests/test_interfaces.py

```diff
@@ -191,6 +191,36 @@ async def test_http_proxy(
         assert ext == {"http_version": "HTTP/1.1", "reason": "OK"}
 
 
+@pytest.mark.parametrize("proxy_mode", ["DEFAULT", "FORWARD_ONLY", "TUNNEL_ONLY"])
+@pytest.mark.parametrize("protocol,port", [(b"http", 80), (b"https", 443)])
+@pytest.mark.trio
+async def test_proxy_socket_does_not_leak_when_the_connection_hasnt_been_added_to_pool(
+    proxy_server: URL,
+    server: Server,
+    proxy_mode: str,
+    protocol: bytes,
+    port: int,
+):
+    method = b"GET"
+    url = (protocol, b"example.com", port, b"/")
+    headers = [(b"host", b"example.org")]
```

Comment:
Is the host mismatch between url / headers expected? I think this is somewhat related to the _block file_ passed to `pproxy`. If so, ad assuming the block file is intended to be a list of URL hosts that _cannot_ be requested through the proxy…

* Should we set the blocked hostname to something more explicit like `blockedhost.org`?
* If the different host in `url` and `headers` is on purpose and required, add a small comment to explain why?

Reply:
You are right! I missed that the url and header are different, and it might confuse. Moreover the server name is not so clear. I'm renaming this host to "blockedhost.example.com" (and changing the fixture the same way)